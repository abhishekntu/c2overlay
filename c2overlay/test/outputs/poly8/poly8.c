/****************************************************************/
/*   poly8		                                        */
/****************************************************************/

#ifdef TEST
#include <stdio.h>
#endif

#include<math.h>

#ifdef ALGO
int foo(int t,int y,int z)
{	
int t1_1,t1_2,t1_3,t1_4,t1_5,t1_6,t1_7,t1_8,t1_9,t1_10,t2_1,t2_2,t2_3,t2_4,t2_5,t2_6,t2_7,t2_8,t3_1,t3_2,t3_3,t3_4,t3_5,t3_6,t3_7,t4_1,t4_2,t4_3,t4_4,t5_1,t5_2,t5_3,t6_1,t6_2,t6_3,t7_1,t7_2,t8_1,t8_2,t9_1,t10_1,t11_1;

t1_1 = y - z;
t1_2 = 71*z;
t1_3 = 464*z;
t1_4 = 4*y + 288;
t1_5 = 360*z;
t1_6 = 6*z;
t1_7 = 432*y;
t1_8 = z - 13824;
t2_1 = y*t1_1;
t2_2 = t1_2 + 1728;
t2_3 = t1_3 - 13824;
t2_4 = y*t1_4;
t2_5 = t1_5 + 6912;
t2_6 = t1_6 - 4312;
t2_7 = t1_7 - 13824;
t2_8 = z*t1_8;
t3_1 = t2_1 + t2_2;
t3_2 = t2_4 + t2_5;
t3_3 = z*t2_6;
t3_4 = z*t2_7;
t4_1 = y*t3_1;
t4_2 = y*t3_2;
t4_3 = t3_3 + 55296;
t5_1 = t4_1 - t2_3;
t5_2 = t4_2 + t4_3;
t6_1 = t*t5_1;
t7_1 = t6_1 + t5_2;
t8_1 = t*t7_1;
t9_1 = t8_1 + t3_4;
t10_1 = t*t9_1;
return(t10_1 + t2_8);


}
#endif


#ifdef TEST
int main(void)
{
	int t,y,z;
 	int result;
  	result = foo(t,y,z);
  	return 0;
}
#endif
