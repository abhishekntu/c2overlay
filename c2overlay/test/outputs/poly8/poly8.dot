digraph graphname {
N22 [color="black", label="O0_N36_N22"];
N21 [color="black", label="sub_Imm_13824_N34_N21"];
N20 [color="black", label="N31N33_mul_N31_add_Imm_55296_N33_N20"];
N8 [color="black", label="N9N32_mul_N9_add_N32_N8"];
N9 [color="black", label="N4N17_mul_N4_add_N17_N9"];
N1 [color="black", label="I0_N1_N1"];
N2 [color="black", label="I1_N2_N2"];
N3 [color="black", label="I2_N3_N3"];
N4 [color="black", label="N28N23_mul_Imm_360_N28_add_Imm_6912_N23_N4"];
N5 [color="black", label="N21N26_mul_Imm_432_N21_sub_Imm_13824_N26_N5"];
N6 [color="black", label="N27N20_mul_N27_add_N20_N6"];
N7 [color="black", label="mul_N8_N7"];
N12 [color="black", label="N7N15_mul_N7_sub_N15_N12"];
N13 [color="black", label="N13N29_mul_Imm_464_N13_sub_Imm_13824_N29_N13"];
N10 [color="black", label="N5N12_add_Imm_72_N5_mul_Imm_4_N12_N10"];
N11 [color="black", label="N6N18_mul_N6_add_N18_N11"];
N16 [color="black", label="N16N24_mul_N16_add_N24_N16"];
N17 [color="black", label="N14N22_mul_Imm_71_N14_add_Imm_1728_N22_N17"];
N14 [color="black", label="N10N35_mul_N10_add_N35_N14"];
N15 [color="black", label="N11N25_mul_Imm_6_N11_sub_Imm_4312_N25_N15"];
N18 [color="black", label="sub_N19_N18"];
N19 [color="black", label="mul_N30_N19"];
N21 -> N19;
N20 -> N8;
N8 -> N11;
N9 -> N22;
N1 -> N14;
N1 -> N11;
N1 -> N9;
N2 -> N10;
N2 -> N6;
N2 -> N16;
N2 -> N8;
N2 -> N5;
N2 -> N12;
N2 -> N18;
N3 -> N20;
N3 -> N17;
N3 -> N21;
N3 -> N13;
N3 -> N19;
N3 -> N7;
N3 -> N15;
N3 -> N4;
N3 -> N18;
N4 -> N6;
N5 -> N7;
N6 -> N8;
N7 -> N14;
N12 -> N11;
N13 -> N12;
N10 -> N6;
N11 -> N14;
N16 -> N12;
N17 -> N16;
N14 -> N9;
N15 -> N20;
N18 -> N16;
N19 -> N9;
}
