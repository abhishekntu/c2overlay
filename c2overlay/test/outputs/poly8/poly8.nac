procedure foo(in s32 t,in s32 y,in s32 z,out s32 D_2448)
{
  localvar s32 D_2447;
  localvar s32 t1_1;
  localvar s32 t1_2;
  localvar s32 t1_3;
  localvar s32 t1_4;
  localvar s32 t1_5;
  localvar s32 t1_6;
  localvar s32 t1_7;
  localvar s32 t1_8;
  localvar s32 t1_9;
  localvar s32 t1_10;
  localvar s32 t2_1;
  localvar s32 t2_2;
  localvar s32 t2_3;
  localvar s32 t2_4;
  localvar s32 t2_5;
  localvar s32 t2_6;
  localvar s32 t2_7;
  localvar s32 t2_8;
  localvar s32 t3_1;
  localvar s32 t3_2;
  localvar s32 t3_3;
  localvar s32 t3_4;
  localvar s32 t3_5;
  localvar s32 t3_6;
  localvar s32 t3_7;
  localvar s32 t4_1;
  localvar s32 t4_2;
  localvar s32 t4_3;
  localvar s32 t4_4;
  localvar s32 t5_1;
  localvar s32 t5_2;
  localvar s32 t5_3;
  localvar s32 t6_1;
  localvar s32 t6_2;
  localvar s32 t6_3;
  localvar s32 t7_1;
  localvar s32 t7_2;
  localvar s32 t8_1;
  localvar s32 t8_2;
  localvar s32 t9_1;
  localvar s32 t10_1;
  localvar s32 t11_1;
L0005:
  t1_1 <= sub y,z;
  t1_2 <= mul z,71;
  t1_3 <= mul z,464;
  D_2447 <= add y,72;
  t1_4 <= mul D_2447,4;
  t1_5 <= mul z,360;
  t1_6 <= mul z,6;
  t1_7 <= mul y,432;
  t1_8 <= sub z,13824;
  t2_1 <= mul y,t1_1;
  t2_2 <= add t1_2,1728;
  t2_3 <= sub t1_3,13824;
  t2_4 <= mul y,t1_4;
  t2_5 <= add t1_5,6912;
  t2_6 <= sub t1_6,4312;
  t2_7 <= sub t1_7,13824;
  t2_8 <= mul z,t1_8;
  t3_1 <= add t2_1,t2_2;
  t3_2 <= add t2_4,t2_5;
  t3_3 <= mul z,t2_6;
  t3_4 <= mul z,t2_7;
  t4_1 <= mul y,t3_1;
  t4_2 <= mul y,t3_2;
  t4_3 <= add t3_3,55296;
  t5_1 <= sub t4_1,t2_3;
  t5_2 <= add t4_2,t4_3;
  t6_1 <= mul t,t5_1;
  t7_1 <= add t6_1,t5_2;
  t8_1 <= mul t,t7_1;
  t9_1 <= add t8_1,t3_4;
  t10_1 <= mul t,t9_1;
  D_2448 <= add t10_1,t2_8;
}
