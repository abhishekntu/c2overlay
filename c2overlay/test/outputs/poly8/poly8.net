.input N1
pinlist: N1
    
.input N2
pinlist: N2
    
.input N3
pinlist: N3
    
.output out:N22
pinlist: N22
    
.clb N21_blk
pinlist: N3 open open open N21 open open open open 
subblock: N21_blk 0 open open open 4 open open open open

.clb N20_blk
pinlist: N3 N15 open open N20 open open open open 
subblock: N20_blk 0 1 open open 4 open open open open

.clb N8_blk
pinlist: N6 N2 N20 open N8 open open open open 
subblock: N8_blk 0 1 2 open 4 open open open open
    
.clb N9_blk
pinlist: N1 N14 N19 open N22 open open open open 
subblock: N9_blk 0 1 2 open 4 open open open open
    
.clb N4_blk
pinlist: N3 open open open N4 open open open open 
subblock: N4_blk 0 open open open 4 open open open open

.clb N5_blk
pinlist: N2 open open open N5 open open open open 
subblock: N5_blk 0 open open open 4 open open open open

.clb N6_blk
pinlist: N4 N2 N10 open N6 open open open open 
subblock: N6_blk 0 1 2 open 4 open open open open
    
.clb N7_blk
pinlist: N5 N3 open open N7 open open open open 
subblock: N7_blk 0 1 open open 4 open open open open

.clb N12_blk
pinlist: N2 N13 N16 open N12 open open open open 
subblock: N12_blk 0 1 2 open 4 open open open open
    
.clb N13_blk
pinlist: N3 open open open N13 open open open open 
subblock: N13_blk 0 open open open 4 open open open open

.clb N10_blk
pinlist: N2 open open open N10 open open open open 
subblock: N10_blk 0 open open open 4 open open open open

.clb N11_blk
pinlist: N8 N1 N12 open N11 open open open open 
subblock: N11_blk 0 1 2 open 4 open open open open
    
.clb N16_blk
pinlist: N2 N17 N18 open N16 open open open open 
subblock: N16_blk 0 1 2 open 4 open open open open
    
.clb N17_blk
pinlist: N3 open open open N17 open open open open 
subblock: N17_blk 0 open open open 4 open open open open

.clb N14_blk
pinlist: N7 N1 N11 open N14 open open open open 
subblock: N14_blk 0 1 2 open 4 open open open open
    
.clb N15_blk
pinlist: N3 open open open N15 open open open open 
subblock: N15_blk 0 open open open 4 open open open open

.clb N18_blk
pinlist: N2 N3 open open N18 open open open open 
subblock: N18_blk 0 1 open open 4 open open open open

.clb N19_blk
pinlist: N3 N21 open open N19 open open open open 
subblock: N19_blk 0 1 open open 4 open open open open

