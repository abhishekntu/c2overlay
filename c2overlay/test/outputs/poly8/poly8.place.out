Netlist file: test.net   Architecture file: test.arch.xml
Array size: 5 x 5 logic blocks

#block name	x	y	subblk	block number
#----------	--	--	------	------------
N1		0	2	0	#0
N2		2	0	0	#1
N3		6	4	0	#2
out:N22		0	3	0	#3
N21_blk		1	1	0	#4
N20_blk		4	3	0	#5
N8_blk		3	3	0	#6
N9_blk		1	3	0	#7
N4_blk		4	4	0	#8
N5_blk		2	1	0	#9
N6_blk		3	4	0	#10
N7_blk		2	2	0	#11
N12_blk		4	2	0	#12
N13_blk		5	2	0	#13
N10_blk		2	4	0	#14
N11_blk		3	2	0	#15
N16_blk		4	1	0	#16
N17_blk		5	1	0	#17
N14_blk		2	3	0	#18
N15_blk		5	3	0	#19
N18_blk		3	1	0	#20
N19_blk		1	2	0	#21
